architecture/index.md

```diff
@@ -39,6 +39,55 @@ API resources in the second row (PipelineRun, Snapshot) should primarilly be tho
 data-plane resources. The system responds to user requests by creating and managing the lifecycle of
 these resources.
 
+- An [Application] represents a functionally coherent set of [Components] that should be built,
+  tested, and released together. The user provides and names their [Applications]. They are
+  generally long-lived and don't change much after they are created.
+- A [Component] represents an OCI artifact to build from a particular git branch (and a particular context
+  directory). It is owned by an [Application]. The user provides and names their [Components] and
+  specifies which git repositories and branches define how the artifact is built. The user may add,
+  remove, or change [Components] over the lifespan of an [Application].
+- A [Snapshot] represents a collection of particular OCI artifact, mapped to the [Components] of an
```

Comment:
It seems like "OCI artifact" is redefined here vs. line 45. I think I get the idea that this is a particular build of the artifact, but I would assume Component branches may change over time without the user redefining the Component, and those would result in multiple of these output OCI artifacts. If Snapshot is the output of building / assembling an Application from a particular set of Component builds, is there a term for the built outputs of those Components?

Reply:
> If Snapshot is the output of building / assembling an Application from a particular set of Component builds, is there a term for the built outputs of those Components?

No - there's no formal resource on-cluster that represents that. Informally, I've heard it called a "build" or an "image". It's an OCI artifact, described by a pullspec pinned to digest.

Reply:
Is it fair to say that the system assumes everything will be expressed in OCI terms? I wonder if that's a safe assumption for the long term? (That's a discussion for elsewhere)

Reply:
Yes, I think that the system currently assumes that any artifact would be wrapped up in an OCI artifact. One of the key drivers for this assumption is the behavior of Tekton Chains being able to create a signed provenance attestation for the artifact. Expanding to additional artifact storage formats/strategies would likely require some rearchitecting.